### PR TITLE
Replace call to profile.get_refresh_token() with profile.get_raw_token()

### PR DESCRIPTION
**Pull request by BrianBlum**
Only internal method call change.
---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/azure-cli/azure/cli/command_modules/ams/operations/mru.py [hunk 1]
```diff
@@ -9,7 +9,13 @@
...
 from azure.cli.core.util import CLIError
 from azure.cli.command_modules.ams._completers import get_mru_type_completion_list
-
+from azure.cli.core.commands.client_factory import get_subscription_id
+from azure.cli.core._profile import _ACCESS_TOKEN
+from azure.cli.core._profile import _REFRESH_TOKEN
+from azure.cli.core._profile import _SERVICE_PRINCIPAL
+from azure.cli.core._profile import _USER_ENTITY
+from azure.cli.core._profile import _USER_TYPE
+from azure.cli.core._profile import _USER
```
Comment (tjprescott) [inline src/azure-cli/azure/cli/command_modules/ams/operations/mru.py]: These should just be combined:
```Python
from azure.cli.core._profile import (
_ACCESS_TOKEN, _REFRESH_TOKEN, ...)
```

Comment (BrianBlum) [reply src/azure-cli/azure/cli/command_modules/ams/operations/mru.py]: I'm using those profile constants in the results of the profile calls. If I just replicate the strings in my code, and profile changes a string, I would definitely break.

File: src/azure-cli/azure/cli/command_modules/ams/operations/mru.py [hunk 2]
```diff
@@ -84,24 +90,34 @@ def _get_v2_api_endpoint(self, cli_ctx, resource_group_name, account_name):
...
+            refresh_token = raw_token[_REFRESH_TOKEN]
+            return context.acquire_token_with_refresh_token(refresh_token,
+                                                            clientId,
+                                                            self.v2_media_api_resource).get(_ACCESS_TOKEN)
+        if user_type == _SERVICE_PRINCIPAL:
+            sp_authinfo = self.profile.get_sp_auth_info()
+            acq_token = context.acquire_token_with_client_credentials(self.v2_media_api_resource,
+                                                                      sp_authinfo['clientId'],
+                                                                      sp_authinfo['clientSecret'])
+            return acq_token['accessToken']
```
Comment (yugangw-msft) [inline src/azure-cli/azure/cli/command_modules/ams/operations/mru.py]: I believe the logics here can be replaced with one line call of `get_row_token(resource=self.v2_media_api_resource)`

Comment (BrianBlum) [reply src/azure-cli/azure/cli/command_modules/ams/operations/mru.py]: This statement works for both user logins and service principal: self.profile.get_raw_token(resource=self.v2_media_api_resource)[0][2].get(_ACCESS_TOKEN)
Do you recommend **NOT** using _ACCESS_TOKEN from _profile.py?

File: src/azure-cli/azure/cli/command_modules/ams/operations/mru.py [hunk 3]
```diff
@@ -55,15 +56,15 @@ def __init__(self, cli_ctx, resource_group_name, account_name):
...
     def _get_v2_api_endpoint(self, cli_ctx, resource_group_name, account_name):
         from msrestazure.tools import resource_id
         from azure.cli.command_modules.ams._sdk_utils import (get_media_namespace, get_media_type)
-        from azure.cli.core.commands.client_factory import get_subscription_id
 
-        refresh_token_obj = self.profile.get_refresh_token()
-        access_token = refresh_token_obj[2]
+        raw_token_obj = self.profile.get_raw_token()
+        raw_token = raw_token_obj[0][2]
+        access_token = raw_token[_ACCESS_TOKEN]
```
Comment (yugangw-msft) [inline src/azure-cli/azure/cli/command_modules/ams/operations/mru.py]: should be `_, access_token, _ = self.profile.get_raw_token()`

File: src/azure-cli/azure/cli/command_modules/ams/operations/mru.py (hunk 2 above)
Comment (tjprescott) [reply src/azure-cli/azure/cli/command_modules/ams/operations/mru.py]: You should not take a dependency on anything that is private (leading _) as it explicitly makes no guarantee of stability.

Comment (BrianBlum) [reply src/azure-cli/azure/cli/command_modules/ams/operations/mru.py]: @tjprescott Thanks Travis. I'll use a local string.

**Comment by BrianBlum** [comment]
@tjprescott Thanks Travis.

**Comment by BrianBlum** [comment]
@tjprescott How do I find which Milestone this fix is going in? We have a customer waiting for sp login support.
Thanks

**Comment by tjprescott** [comment]
https://github.com/Azure/azure-cli/milestones

**Pull request merged**